engine/errors.go

```diff
@@ -46,6 +46,38 @@ func IsInvalidInputError(err error) bool {
 	return errors.As(err, &errInvalidInputError)
 }
 
+// IsNetworkTransmissionError returns whether the given error is a NetworkTransmissionError error
+func IsNetworkTransmissionError(err error) bool {
+	var errNetworkTransmissionError NetworkTransmissionError
+	return errors.As(err, &errNetworkTransmissionError)
+}
+
+// NetworkTransmissionError captures the general sentinel errors upon network transmission. It is used to distinguish
+// network transmission errors from other errors.
+type NetworkTransmissionError struct {
+	err error
+}
```

Comment:
Thank you for adding this. This is really useful, given that the network layer doesn't have error returns defined. 

If we don't already have one, it would be great to create a tech debt issue to eventually replace this with a similar error (or errors) which are returned directly directly by `network.Conduit`. That way we can:
* detect unexpected failure cases from calls to `Conduit.*` functions
* define errors in `Conduit` implementation once, rather than wrapping at all `Conduit.*` call-sites

Reply:
Added a place holder issue https://github.com/dapperlabs/flow-go/issues/6418 and will scope it shortly.